.dockerignore

```diff
@@ -0,0 +1,2 @@
+**
+!entrypoint.sh
```

Comment:
I have mixed feelings with this.

There are people that prefer to have a `dockerignore` that ignores everything so you are sure on what you want to have in your Docker context. It means that every time that you have to add a file to the context you have to add it to the `dockerignore` so it is included.

On the other hand, there are people that prefer to create a folder like `build` where you leave the binaries build (not in the case of this project) and you send the whole `build` as docker contexts, with no `dockerignore`. So the `Dockerfile` should shit in the `build` folder too. As a con to this approach, if you have a file that is shared at compile time and build time, the file has to be duplicated of you have to point at compilation to the `build` folder that could seem and anti-pattern.

I really prefer the second one, because the first one is prone to forget that the `.dockeignore` exists (as it does not appear in `ls -lh`) and can be cause of footgun.

But I would love to know your thoughts about this.

Reply:
I've been footgunned by `.dockerignore` once, that's why I added a comment in the `Dockerfile`:
https://github.com/txqueuelen/stateless-dns/blob/3e007616e44a99839477dcee54caf36db8fcd332/Dockerfile#L20

I have a slight preference for the ignore approach because it does not "hide" the dockerfile away, but I can change it if you think the comment is not enough.

Reply:
It is not if the comment is enough or not.

It is the preference decission:
 * Having a "dirty repo" with lots of things in the root and choosing what goes to the docker context or not.
 * Having a "dirty repo" with lots of things in the root but have a clean folder with everything that should go to the image.
 
 As repositories tend to have lots of things in the root (like READMEs, Changelogs, configurations for linters and IDEs, yada, yada,...) I am hesitant of having `Dockerfiles` at root too. For me is hard to navigate into...

Reply:
Funny because when I'm glancing at random projects at github for selfhosting, something I do look out for are Dockerfiles in the root folder 😆 

But again, no strong pref, so I'll go for the build forlder.

Reply:
> Funny because when I'm glancing at random projects at github for selfhosting, something I do look out for are Dockerfiles in the root folder 😆

🤔

Reply:
I am taking a second thought on this and I think you are right. Maybe having it at root make it way easy to look for it...

Reply:
To the root it goes!